### PR TITLE
Add secret tree test vector

### PR DESCRIPTION
**Pull request by mulmarta**
Depends on #72

### CODE REVIEW

File: test-vectors/secret-tree.json [hunk 1]
```diff
@@ -0,0 +1,2284 @@
...
+      "nonce": "f604e6e95af3536e722fc18d"
+    },
+    "leaves": [
+      {
+        "generation": 15,
+        "application_key": "2295064552a7c387bcb2902876f1a6cb",
+        "application_nonce": "4dd0a464188d522c50211d59",
+        "handshake_key": "8e82d0c1ad6f44ac65465658be9365f4",
+        "handshake_nonce": "78f0edef08fc96cb6a030e42"
+      },
```
Comment (bifurcation) [inline test-vectors/secret-tree.json:39]: You should have an array of generations for each leaf not just a single generation.

**Review by bifurcation — CHANGES_REQUESTED** [review]
I reviewed for structure, not contents.  Overall, looks correct to me, except for noted problem.

File: test-vectors/secret-tree.json [hunk 2]
```diff
@@ -0,0 +1,2284 @@
+[
+  {
+    "cipher_suite": 1,
+    "n_leaves": 1,
```
Comment (bifurcation) [inline test-vectors/secret-tree.json:4]: This is no longer necessary; you can just use the length of the `leaves` array.

**Comment by bifurcation** [comment]
FWIW, I just took a stab at [implementing this in MLSpp](https://github.com/cisco/mlspp/pull/299), and this is what my output looks like:

``` json
[
  {
    "cipher_suite": 1,
    "sender_data": {
      "ciphertext": "2bf6eff289b2127f13678f335304625d0fb185d6916a839422816eff3edad135",
      "key": "901fcfea8808b97277c1727c948d75a8",
      "nonce": "c2a5fc1adce345a1d3ff0c4e",
      "sender_data_secret": "797b08581ca1c8ed189061ea26e3379acf4d36f74a58ec978c317c860c8e2c57"
    }
    "encryption_secret": "3dceb4d43c0e59cfd8c1442c09de0a63884938c5cd91b48cd1f020b6e3683291",
    "leaves": [
      [
        {
          "application_key": "43b83df88fa5676eda6bafc01303d1be",
          "application_nonce": "db2c0ea31307b012741b6d32",
          "generation": 1,
          "handshake_key": "6bd650bd814c61783fdedad0bc2138b4",
          "handshake_nonce": "52334c0a7c5bb5225d9c206d"
        },
        {
          "application_key": "35cee43767c0d3eb33c19d44aa29152a",
          "application_nonce": "ca9e99bc5ff094435ed08bf6",
          "generation": 15,
          "handshake_key": "46d57b2a274f8248779081947928003e",
          "handshake_nonce": "89eda4f3017d226dfe42e0b8"
        }
      ],
      ...
    ],
  },
  ...
]
```

**Comment by mulmarta** [comment]
I definitely misread the description of the test vector. Sorry about that. Should be good now.

**Review by bifurcation — APPROVED** [review]

**Pull request merged**